Sources/TuistGraph/Graph/GraphOutputJSON.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import TSCBasic
+
+public struct GraphOutputJSON: Codable, Equatable {
+    public var name: String
+    public var path: String
+    public var projects: [String: ProjectOutputJSON]
+    
+    public init(name: String, path: String, projects: [String: ProjectOutputJSON]) {
+        self.name = name
+        self.path = path
+        self.projects = projects
+    }
+    
+    public static func from(_ graph: Graph, withConfig config: Config) -> GraphOutputJSON {
+        var projects = [String: ProjectOutputJSON]()
+        for (_, project) in graph.projects {
+            // Convert packages.
+            var packages = [PackageOutputJSON]()
+            for package in project.packages {
+                switch package {
+                case .remote(let url, _):
+                    packages.append(PackageOutputJSON(kind: PackageOutputJSON.PackageKind.remote, path: url))
+                case .local(let path):
+                    packages.append(PackageOutputJSON(kind: PackageOutputJSON.PackageKind.local, path: path.pathString))
+                }
+            }
+            
+            // Convert custom schemes. These are the explicitly specified schemes not the default
+            // ones generated automatically for all the targets.
+            var schemes = [SchemeOutputJSON]()
+            for scheme in project.schemes {
+                var testTargets = [String]()
+                if let testAction = scheme.testAction {
+                    for testTarget in testAction.targets {
+                        testTargets.append(testTarget.target.name)
+                    }
+                }
+                schemes.append(SchemeOutputJSON(name: scheme.name, testActionTargets: testTargets))
+            }
+            
+            // Convert targets.
+            let didDisableAutogeneratedSchemes = config.disableAutogeneratedSchemes
+            var targets = [TargetOutputJSON]()
+            for target in project.targets {
+                targets.append(TargetOutputJSON(name: target.name, product: target.product.rawValue))
+                
+                // For each target, a default scheme is generated. Include these in the schemes output.
+                // If the target's product is a testing one, include the target as test action.
+                if !didDisableAutogeneratedSchemes {
```

Comment:
I'm not sure if I completely follow this one. Should not autogenerated schemes be part of the graph already? Ideally, this `from` function should not contain a lot of logic like this as it would be then coupled to logic in mappers which will make it easy to make mistakes.

Reply:
That doesn't seem to be the case actually. When I ran this command on the tuist project, it did not output the schemes.

I didn't want to change the mapping logic since I'd imagine that's used in far more places. There could be a legit reason why they are not in the `Graph` instance.

Reply:
I'll just remove this for now then. Let's first get to a point where the output schema matches the current `Graph` type. Later we can change/fix anything if needed.